src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+    test(`should render with shouldCoverButton prop set to ${coverButton}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Box height="330px">
+          <PopoverContainerComponentCoverButton
+            shouldCoverButton={coverButton}
+          />
+        </Box>
+      );
+
+      const popoverContainerContentElement = popoverContainerContent(page);
+      await expect(popoverContainerContentElement).toBeInViewport();
+
+      const positionTop = await getTopValue(popoverContainerContentElement);
+      const positionBottom = await getBottomValue(
+        popoverContainerContentElement
+      );
+      const positionY = await getYValue(popoverContainerContentElement);
+
+      await expect(positionBottom).toBeLessThan(bottomValueMin);
+      await expect(positionTop).toBeLessThan(yAndTopValueMin);
+      await expect(positionY).toBeLessThan(yAndTopValueMin);
+    });
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should open using ${key} keyboard key`, async ({ mount, page }) => {
+      await mount(
+        <PopoverContainer title="String is awesome">Contents</PopoverContainer>
+      );
+
+      const popoverSettingsIconElement = popoverSettingsIcon(page);
+      await popoverSettingsIconElement.press(key);
+
+      await expect(popoverContainerContent(page)).toBeVisible();
+    });
+  });
+
+  ([...keyToTrigger] as const).forEach((key) => {
+    test(`should close using ${key} keyboard key`, async ({ mount, page }) => {
+      await mount(
+        <PopoverContainer title="String is awesome">Contents</PopoverContainer>
+      );
+
+      const popoverSettingsIconElement = popoverSettingsIcon(page);
+      await popoverSettingsIconElement.click();
+      const popoverCloseIconElement = popoverCloseIcon(page);
+      await popoverCloseIconElement.press(key);
+
+      await expect(popoverContainerContent(page)).not.toBeVisible();
+    });
+  });
+
+  test("should close using escape keyboard key", async ({ mount, page }) => {
+    await mount(
+      <PopoverContainer title="String is awesome">Contents</PopoverContainer>
+    );
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+    const popoverContainerContentElement = popoverContainerContent(page);
+    await popoverContainerContentElement.press("Escape");
+
+    await expect(popoverContainerContentElement).not.toBeVisible();
+  });
+
+  test("should not close when an option is selected from Select component inside", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<PopoverContainerWithSelect />);
+
+    await popoverSettingsIcon(page).click();
+    await selectText(page).click();
+    await selectListText(page, "green").click();
+
+    await expect(popoverContainerContent(page)).toBeVisible();
+  });
+
+  // Due to a bug in Playwright whereby the Escape key fires a keyup event before the keydown event.
```

Comment:
**praise**: great comment to explain exactly why we're skipping the test and what needs to happen before we re-enable it.

It does give rise to a **question** though - since it seems 1.40.0 has been released, and I assume (perhaps foolishly!) it shouldn't have any major breaking changes from what we're currently using (1.36.2), is there a good reason not to just bite the bullet and upgrade to take the fix as part of this PR?